test-suite/buildtests/layer-02-maximus.opts

```diff
@@ -111,7 +111,7 @@ DISTCHECK_CONFIGURE_FLAGS=" \
 	--with-pthreads \
 	--enable-build-info=squid\ test\ build \
 	--enable-ssl-crtd \
-	--with-openssl \
+	--with-openssl=${with_openssl:-yes} \
```

Comment:
> The actual underlying issue is that --with-openssl[=location] acts both as an enable switch and as a location pointer (with --with-openssl and --with-openssl=yes being synonyms for "enable, includes and headers are in custom location")

I believe the following is a fairly accurate summary of what --with-openssl variants mean (bugs notwithstanding):

1. --with-openssl: Enable OpenSSL support. Use OpenSSL libraries and headers located using supported-by-Squid discovery mechanisms such as pkg-config and LIBOPENSSL_LIBS.
2. --with-openssl=yes: Exactly the same as above.
3. --with-openssl=path: Enable OpenSSL support. Use OpenSSL libraries and headers located at the given path.

Please correct me if any of the above items are wrong. Being precise about this may help find a correct solution for the problem this PR wants to solve.

> A system-specific workaround is to pass the location in with the $with_openssl environment variable

I think I understand why your workaround works. The essence of my concern is not the mechanics of the workaround. It is that the proposed solution (i.e., `test-builds.sh` caller setting `$with_openssl`) appears to use a _private_ ./configure.ac variable (i.e., `$with_openssl`) to solve the problem at hand (i.e., supplying a custom OpenSSL directory to scripts that call ./configure) where there exist standard and/or  _public_ interfaces that were designed to solve the same problem (i.e., `PKG_CONFIG_PATH` and `LIBOPENSSL_*`).

> LIBOPENSSL_CFLAGS and LIBOPENSSL_LIBS seem to be just duplicates --with-openssl

I am not sure two variables with different meaning can duplicate a single option, but the more important problem here is that `LIBOPENSSL_CFLAGS`, `LIBOPENSSL_LIBS`, and `PKG_CONFIG_PATH` are public interfaces while `$with_openssl` is not. If possible, we should use public interfaces. I do not know why that is impossible in this particular case (as this PR appears to imply).

Could you please clarify why a `test-builds.sh` caller on Homebrew-enabled MacOS cannot set `PKG_CONFIG_PATH` (ideally) or `LIBOPENSSL_CFLAGS` and `LIBOPENSSL_LIBS` (if needed) instead of setting `$with_openssl` and then using that variable when specifying ./configure options?